### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by samiullahbahadur**
### **Project requirements**

- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe
- **User index page:**
- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show pag
- **user show page:**
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.

- **User post index page:**
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

- **Post show page:**

- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @post3 = Post.create(title: 'Post 3', text: 'This is Post 3', user_id: @user.id)
+    @post4 = Post.create(title: 'Post 4', text: 'This is Post 4', user_id: @user.id)
+
+    visit "/users/#{@user.id}"
+  end
+
+  it 'I can see the user\'s profile picture.' do
+    # user = User.create(name: 'sami', photo: 'https://unsplash.com/6519861_z.jpg', bio: 'A teacher')
+
+    # visit "/users/#{user.id}"
```
Comment (RolandM99) [inline spec/features/user_show_spec.rb]: - [ ] Kindly, remove the commented code from your codebase to make your codes clean.

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  it 'I can see the user\'s username.' do
+    # user = User.create(name: 'sami', photo: 'https://unsplash.com/6519861_z.jpg', bio: 'A teacher')
+
+    # visit "/users/#{user.id}"
+    expect(page).to have_content('sami')
+  end
+
+  it 'I can see the number of posts the user has written.' do
+    # user = User.create(name: 'sami', photo: 'https://unsplash.com/6519861_z.jpg', bio: 'A teacher')
+    # Post.create(title: 'My first post', text: 'This is my first post', user_id: user.id)
```
Comment (RolandM99) [inline spec/features/user_show_spec.rb]: - [ ] Remove commented codes from your codebase to make it clean

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi  @samiullahbahadur, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [ ] [**REQUIRED next lesson**]  Once you'll learn about `Devise` and you'll implement Login page for your project, Make sure that you add integration specs for your `Login page` as requested in the project requirements.

- [ ] Kindly, make sure that you're solving the N+1 problem when fetching all posts and their comments for a user. you can implement it by adding for eample in your Posts#index action this line of code : `@posts = @user.posts.includes(:comments)`

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/user_show_spec.rb (hunk 1 above)
Comment (samiullahbahadur) [reply spec/features/user_show_spec.rb]: Hi @RolandM99 you right, i have fix it

File: spec/features/user_show_spec.rb (hunk 2 above)
Comment (samiullahbahadur) [reply spec/features/user_show_spec.rb]: I have fix it

**Review by Kingstalux — APPROVED** [review]
Hi @samiullahbahadur ,
# Status: Approved ✔️ ✔️ ✔️ 
download

 

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**